### PR TITLE
Show usage of @typedoc

### PR DESCRIPTION
**Pull request by nathanl**

### CODE REVIEW

File: lib/elixir/pages/typespecs.md [hunk 1]
```diff
@@ -16,9 +17,10 @@ Type specifications (sometimes referred to as *typespecs*) are defined in differ
 
 See the "User-defined types" and "Defining a specification" sub-sections below for more information on defining types and typespecs.
 
-## A simple example
+## Simple Examples
```
Comment (josevalim) [inline lib/elixir/pages/typespecs.md:20]:
```suggestion
## Simple examples
```

**Comment by josevalim** [comment]
Thank you @nathanl!

Instead of documenting the type of the struct, I think it would make more sense to document the struct itself. So I am not sure if we should push in this particular direction?

**Comment by nathanl** [comment]
> I think it would make more sense to document the struct itself

@josevalim I'm not sure what you mean by this. Do you mean you would describe it in the `@moduledoc`?

File: lib/elixir/pages/typespecs.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Elixir comes with a notation for declaring types and specifications. Elixir is a
 Type specifications (sometimes referred to as *typespecs*) are defined in different contexts using the following attributes:
 
   * `@type`
+  * `@typedoc`
```
Comment (wojtekmach) [inline lib/elixir/pages/typespecs.md:11]: I’m not sure it belongs here, this one accepts docs string, all the other ones the typespec "syntax"

**Comment by josevalim** [comment]
I meant you can do `@doc` right above `defstruct`.

**Comment by nathanl** [comment]
@josevalim Ah ok, I wasn't aware you could do that. I removed the part where I used `@typedoc` with the struct.

@wojtekmach I moved `@typedoc` out of the list.

**Comment by nathanl** [comment]
@josevalim As an experiment, here's a throwaway module where I documented the type with **both** a `@typedoc` **and** a`@doc` above the struct for comparison.

```elixir
defmodule AbsintheRelayKeysetConnection.Thing do
  @moduledoc "Such a fancy module"

  @doc """
  This is the @doc above defstruct

  A struct with the following keys:

  - :name is the name and should be unpronounceable
  - :size is the size in cubic furlongs
  """
  defstruct [:name, :size]

  @typedoc """
  This is the typedoc above @type t, which is the struct

  A struct with the following keys:

  - :name is the name and should be unpronounceable
  - :size is the size in cubic furlongs
  """
  @type t :: %__MODULE__{
    name: String.t(),
    size: integer()
  }

  @doc "A useless function"
  def do_stuff(:foo) do
    "FOOOO!"
  end
end
```

`ex_doc` generates this output:

ex_doc_output

So putting `@doc` above the struct creates an entry in the "Functions" section, and `@typedoc` creates an entry in the "Types" section.

Do you prefer to see the struct documented under "Functions" via `@doc`?

**Comment by josevalim** [comment]
> Do you prefer to see the struct documented under "Functions" via @doc?

Yes, exactly!

**Comment by nathanl** [comment]
@josevalim OK, thanks! I added a commit to document that also.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**